### PR TITLE
Fix issue with wrong index.yaml pr

### PR DESCRIPTION
**Pull request by andriisoldatenko**
## Description

Problem
Old version of helm chart during automatic generation of new helm index.yaml inside release pipeline. 
After careful debugging, I realised

```
helm repo index helm-pkg \
    --url "https://github.com/Dynatrace/dynatrace-operator/releases/download/v0.14.2" \
    --merge "./config/helm/repos/stable/index.yaml" 
```

That was a problem with the branch name during creating automatic PR, so we just used the old branch name, which doesn't include `0.14.0` and `0.14.1` inside `index.yaml`. (you can check `create-pull-request/patch`).

Steps to Reproduce

If you open this automatic PR https://github.com/Dynatrace/dynatrace-operator/pull/2354 and scroll to original commit https://github.com/Dynatrace/dynatrace-operator/pull/2354/commits/c149f8e24f779266f5cff7e05d0669ddee5b4dca you can see we removing 0.14.1 and 0.14.0 and adding 0.14.2.

So the fix includes some commit prefixes to existing branch, more details at https://github.com/marketplace/actions/github-action-for-creating-pull-requests docs.

## How can this be tested?

n/a

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -351,6 +351,8 @@ jobs:
         with:
           base: main
           delete-branch: true
+          branch: create-pull-request/update-helm-index
```
Comment (waodim) [inline .github/workflows/release.yaml:354]: isn't the branch called [create-pull-request/patch](https://github.com/Dynatrace/dynatrace-operator/tree/create-pull-request/patch) ?

Comment (andriisoldatenko) [reply .github/workflows/release.yaml:354]: correct, I don't like original branch name, I want to make it more clear for reader and for purpose of the PR.

Comment (waodim) [reply .github/workflows/release.yaml:354]: Ahhh i get it, thanks :+1:

**Review by waodim — APPROVED** [review]
code lgtm

**Review by luhi-DT — APPROVED** [review]

**Pull request merged**